Sources/App/Controllers/Karma/KarmaController+Slack.swift

```diff
@@ -28,17 +21,28 @@ extension KarmaController {
         }
 
         // Do this in the background
-        processKarmaCommand(req, content: content, responseUrl: responseUrl)
+        send(karmaStatuses: process(karmaCommand: content), to: responseUrl, with: req, format: formatter(for: content))
 
         // Respond immediately
         return .ok
     }
 
-    private func processKarmaCommand(_ req: Request, content: Command, responseUrl: String) {
-        let userIds = karmaParser.userIds(from: content.text ?? "")
+    private func process(karmaCommand command: Command) -> () -> Future<[KarmaStatus]> {
```

Comment:
This comment goes for `process(...` and `formatter(...`. It seems to me that you could just return `-> Future[KarmaStatus]` and `-> KarmaStatusResponse` for the second one. I'm not seeing the need to wrap both of these in a closure.

Reply:
Returning anonymous functions is just a quick abstraction mechanism to avoid code duplication. I could have written separate handling chains for the two commands, that would drop the formatter() method, but that just wasn't the way I was going.

Reply:
Im not saying remove process() or formatter(). It just doesn't seem necessary to return closures. I'm not seeing how you are avoiding duplication. 

My thought is to change return types. So `() -> Future<[KarmaStatus]>` becomes `-> Future<[KarmaStatus]>` and `-> ([KarmaStatus]) -> KarmaStatusResponse` becomes `-> KarmaStatusResponse` and you just pass the [KarmaStatus] into formatter as a parameter and then change signature of `send()` to include Command as a parameter.

Reply:
It doesn't make a difference I suppose.. updated.